src/libs/notification/notification.c

```diff
@@ -660,6 +617,65 @@ int unmountGlobalPlugin (KDB * kdb, Plugin * plugin)
 	return 1;
 }
 
+static void pluginsOpenNotification (KDB * kdb, ElektraNotificationCallback callback, void * data)
+{
+	// iterate over global plugins
+	for (int positionIndex = 0; positionIndex < NR_GLOBAL_POSITIONS; positionIndex++)
+	{
+		for (int subPositionIndex = 0; subPositionIndex < NR_GLOBAL_SUBPOSITIONS; subPositionIndex++)
+		{
+			Plugin * plugin = kdb->globalPlugins[positionIndex][subPositionIndex];
+			if (!plugin)
+			{
+				continue;
+			}
+
+			size_t func = elektraPluginGetFunction (plugin, "openNotification");
+			if (!func)
+			{
+				continue;
+			}
+			ElektraNotificationOpenNotification openNotification = (ElektraNotificationOpenNotification)func;
+			openNotification (plugin, callback, data);
+		}
+	}
+}
+
+static void pluginsCloseNotification (KDB * kdb)
+{
+	// iterate over global plugins
+	for (int positionIndex = 0; positionIndex < NR_GLOBAL_POSITIONS; positionIndex++)
+	{
+		for (int subPositionIndex = 0; subPositionIndex < NR_GLOBAL_SUBPOSITIONS; subPositionIndex++)
+		{
+			Plugin * plugin = kdb->globalPlugins[positionIndex][subPositionIndex];
+			if (!plugin)
+			{
+				continue;
+			}
+
+			size_t func = elektraPluginGetFunction (plugin, "closeNotification");
+			if (!func)
+			{
+				continue;
+			}
+			ElektraNotificationCloseNotification closeNotification = (ElektraNotificationCloseNotification)func;
+			closeNotification (plugin);
+		}
+	}
+}
+
+void elektraNotificationDoUpdate (Key * key, void * data)
+{
+	KDB * kdb = (KDB *)data;
```

Comment:
Why is `void*` used instead of `KDB*`?

Yes, it can make sense to have an extra struct, then you can pass more than a `KDB*` handle to the callback.

In any case: This is a question of API design. Why should it be related to security?

Reply:
I haven't decided about the struct but didn't want to use `KDB` so I used void for now.

I thought in our discussion on why `KBD` was not in `Plugin` security was the issue.

If having KDB passed in a struct to plugins is not an issue, I'll use a struct to keep the API easier to maintain.